lib/scope.js

```diff
@@ -2485,10 +2485,35 @@ module.exports = {
       if ( !elem ) { reports.addToReport(scope, { type: `error`, value: msg }); }
       expect( elem, msg ).to.exist;
 
-      scope.toDownload = filename;
-      // Should this be using `scope.tapElement()`?
-      await elem.evaluate( elem => { return elem.click(); });
-      await scope.detectDownloadComplete( scope );
+      failed_to_download = false;
+      try {
+        const binaryStr = await scope.page.evaluate(el => {
+          const url = el.getAttribute("href");
+          return new Promise(async (resolve, reject) => {
+            const response = await fetch(url, {method: "GET"});
+            const reader = new FileReader();
+            reader.readAsBinaryString(await response.blob());
+            reader.onload = () => resolve(reader.result);
+            reader.onerror = () => reject('Error occurred while reading binary string');
+          });
+        }, elem);
+        if (binaryStr !== '') {
+          const fileData = Buffer.from(binaryStr, 'binary');
+          fs.writeFileSync(`${ scope.paths.scenario }/${ filename }`, fileData);
```

Comment:
This is great - we can let the author rename the file now, which might be good if they want to do things like generate baseline files with the right names.